solutions/usaco-895.mdx

```diff
@@ -169,4 +169,72 @@ int main() {
 
 </CPPSection>
 
+<PySection>
+```python
+from collections import deque
+
+with open('perimeter.in') as r:
+	t = r.readline
+	n = int(t())
+	ice = []
+	visited = [[False] * n for _ in range(n)]
+	
+	for _ in range(n):
+		ice.append(list(t()))
+
+max_area = 0
+min_peri = float('inf')
+	
+# possible directions
+DIRECTIONS = [
+	(1, 0),
+	(-1, 0),
+	(0, 1),
+	(0, -1)
+]
+
+# check if out of bounds
+def out(a, b, l):
+	if a < 0 or b < 0 or a >= l or b >= l:
+		return True
+	return False
+
+# bfs
```

Comment:
what's this `bfs`-ing?

Reply:
breadth first search

Reply:
ok uh the function name is fine now, just delete this

Reply:
okay